numba/tests/test_errorhandling.py

```diff
@@ -469,10 +471,54 @@ def foo(x):
                     expected = "object has no attribute 'some_invalid_attr'"
                     self.assertIn(expected, str(raises.exception))
 
-    @TestCase.run_test_in_subprocess(
-        envvars={"NUMBA_CAPTURED_ERRORS": "old_style"},
-    )
-    def test_old_style_deprecation(self):
+    def _run_in_separate_process(self, runcode, env):
+        # Run code in separate process with -Wall and specific env-vars
+        code = f"""if 1:
+            {runcode}\n
+            """
+        # On windows, missing the base environment variable can cause
+        # Fatal Python error: _Py_HashRandomization_Init: failed to get random
+        #                     numbers to initialize Python
+        proc_env = os.environ.copy()
+        proc_env.update(env)
+        popen = subprocess.Popen([sys.executable, "-Wall", "-c", code],
+                                 stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE,
+                                 env=proc_env)
+
+        out, err = popen.communicate()
+        if popen.returncode != 0:
+            raise AssertionError("process failed with code %s: stderr follows"
+                                 "\n%s\n" % (popen.returncode, err.decode()))
+        return out, err
+
+    def test_old_style_deprecation_on_import(self):
+        from numba.core.config import _old_style_deprecation_msg
+
+        code = """
+        import numba
+        """
+        # Check that the deprecated message is shown
+        # if NUMBA_CAPTURED_ERRORS=old_style
+        env = {"NUMBA_CAPTURED_ERRORS": "old_style"}
+        _out, err = self._run_in_separate_process(code, env)
```

Comment:
Can `TestCase.run_test_in_subprocess` be used with the `PYTHONWARNINGS` env var to create the same effect as the above `_run_in_separate_process` with the `-Wall` option?

Reply:
I chose to not use `TestCase.run_test_in_subprocess` because that will use `numba.runtests` which implicitly import numba. It end up making it very hard to debug if something goes wrong.

Reply:
Ah, ok, that makes sense, thanks for explaining.